nestedtensor/nested/masking.py

```diff
@@ -25,45 +22,41 @@ def nested_tensor_from_tensor_mask(tensor, mask, nested_dim=None):
     if not mask.dim() > 0:
         raise RuntimeError("Mask has to have dimention > 0")
 
-    # TODO: Need to define empty-list and 0 numel semantics
-    # TODO: Test edge-cases and kwargs
-    def _merge(tensors, nested_dim=None):
-        assert len(tensors) > 0
+    if nested_dim is not None and mask.dim() < nested_dim:
+        raise ValueError("Mask dimension ({0}) is too small to construct nested tensor with nested dimension of ({1})".format(mask.dim(), nested_dim))
```

Comment:
No, this depends on the dimension of tensor. You can always have nested_dim == dim if all Tensor constituents are scalars. You can use "to_nested_tensor" to unroll a given NestedTensor into a NestedTensor of NestedTensors.